public/components/trace_analytics/components/traces/span_detail_panel.tsx

```diff
@@ -235,70 +348,145 @@ export function SpanDetailPanel(props: {
       id: 'span_list',
       label: 'Span list',
     },
+    {
+      id: 'hierarchy_span_list',
+      label: 'Tree view',
+    },
   ];
   const [toggleIdSelected, setToggleIdSelected] = useState(toggleOptions[0].id);
 
   const spanDetailTable = useMemo(
     () => (
-      <SpanDetailTable
-        http={props.http}
-        hiddenColumns={mode === 'jaeger' ? ['traceID', 'traceGroup'] : ['traceId', 'traceGroup']}
-        DSL={DSL}
-        mode={mode}
-        openFlyout={(spanId: string) => {
-          if (fromApp) {
-            props.openSpanFlyout(spanId);
-          } else {
-            setCurrentSpan(spanId);
-          }
-        }}
-        dataSourceMDSId={props.dataSourceMDSId}
-      />
+      <div style={{ width: dynamicLayoutAdjustment }}>
+        <SpanDetailTable
+          http={props.http}
+          hiddenColumns={mode === 'jaeger' ? ['traceID', 'traceGroup'] : ['traceId', 'traceGroup']}
+          DSL={DSL}
+          mode={mode}
+          openFlyout={(spanId: string) => {
+            if (fromApp) {
+              props.openSpanFlyout(spanId);
+            } else {
+              setCurrentSpan(spanId);
+            }
+          }}
+          dataSourceMDSId={props.dataSourceMDSId}
+        />
+      </div>
+    ),
+    [DSL, setCurrentSpan, dynamicLayoutAdjustment]
+  );
+
+  const spanDetailTableHierarchy = useMemo(
+    () => (
+      <div style={{ width: dynamicLayoutAdjustment }}>
+        <SpanDetailTableHierarchy
+          http={props.http}
+          hiddenColumns={mode === 'jaeger' ? ['traceID', 'traceGroup'] : ['traceId', 'traceGroup']}
+          DSL={DSL}
+          mode={mode}
+          openFlyout={(spanId: string) => {
+            if (fromApp) {
+              props.openSpanFlyout(spanId);
+            } else {
+              setCurrentSpan(spanId);
+            }
+          }}
+          dataSourceMDSId={props.dataSourceMDSId}
+        />
+      </div>
     ),
-    [DSL, setCurrentSpan]
+    [DSL, setCurrentSpan, dynamicLayoutAdjustment]
   );
 
   const ganttChart = useMemo(
     () => (
       <Plt
-        data={data.gantt}
+        data={data.gantt.map((trace) => {
+          const duration = trace.x[0] ? trace.x[0].toFixed(2) : '0.00'; // Format duration to 2 decimal places
+
+          return {
+            ...trace,
+            text: `${duration} ms`,
+            textposition: 'outside',
+            hoverinfo: 'none',
+          };
+        })}
         layout={layout}
         onClickHandler={onClick}
         onHoverHandler={onHover}
         onUnhoverHandler={onUnhover}
+        onRelayout={(event) => {
+          // Handle x-axis range update
+          if (event && event['xaxis.range[0]'] && event['xaxis.range[1]']) {
+            const newRange = [event['xaxis.range[0]'], event['xaxis.range[1]']];
+            setSelectedRange(newRange);
+          } else {
+            setSelectedRange(fullRange);
+          }
+        }}
```

Comment:
Nit - not blocking. Just for good readability do you want to follow how other props functions are declared and used here. Something like `onClick`.